library/Translate/src/Juvix/Contextify/ToContext/Sexp.hs

```diff
@@ -119,67 +129,33 @@ declare (Sexp.List [inf, n, i]) ctx
                   Context.Right
                 | otherwise -> error "malformed declaration"
             (fromIntegral atomNum)
-        meta = lookupInfoTable atomName ctx
-     in pure $ case Context.extractValue <$> Context.lookup (pure atomName) ctx of
-          Just (Context.Info _ (Context.Def d)) ->
-            Type.PS
-              { ctxS =
-                  ctx
-                    |> Context.add
-                      (NameSpace.Pub atomName)
-                      ( Context.Info
-                          meta
-                          (Context.Def (d {Context.defPrecedence = prec}))
-                      ),
-                opensS = [],
-                modsDefinedS = []
-              }
-          _ ->
-            Type.PS
-              { ctxS =
-                  Context.add
-                    (NameSpace.Pub atomName)
-                    (Context.Info meta (Context.Information [Context.Prec prec]))
-                    ctx,
-                opensS = [],
-                modsDefinedS = []
-              }
+        newCtx =
+          injectMetaInformation ctx atomName ("precedence", Sexp.serialize prec)
+     in Type.PS newCtx [] [] |> pure
 declare _ _ = error "malformed declare"
 
 -- | @type'@ will take its type and add it into the context. Note that
 -- since we store information with the type, we will keep the name in
 -- the top level form.
-type' :: Sexp.T -> Context.T Sexp.T Sexp.T Sexp.T -> IO Type.PassSexp
+type' :: Sexp.T -> Context.T -> IO Type.PassSexp
 type' t@(assocName Sexp.:> _ Sexp.:> dat) ctx
   | Just name <- eleToSymbol (Sexp.car assocName) =
     let constructors = collectConstructors dat
-        meta = lookupInfoTable name ctx
-        addSum con =
-          Context.Sum Nothing name
-            |> Context.SumCon
-            |> Context.Info meta
-            |> Context.add (NameSpace.Pub con)
+        addSum con ctx =
+          Structure.SumCon (pure name)
+            |> Structure.fromSumCon
+            |> injectNewTerm ctx con
```

Comment:
This allows the sum constructors to have infix information, as this does not override such info